nucypher/cli/actions.py

```diff
@@ -405,42 +406,38 @@ def select_stake(stakeholder, emitter, divisible: bool = False, staker_address:
 
 def select_client_account(emitter,
                           provider_uri: str = None,
-                          wallet=None,
+                          signer_uri: str = None,
+                          wallet: Wallet = None,
                           prompt: str = None,
                           default: int = 0,
                           registry=None,
-                          show_balances: bool = True,
+                          show_eth_balance: bool = False,
+                          show_nu_balance: bool = False,
                           show_staking: bool = False,
                           network: str = None,
                           poa: bool = None
                           ) -> str:
     """
-    Note: Setting show_balances to True, causes an eager contract and blockchain connection.
+    Note: Showing ETH and/or NU balances, causes an eager blockchain connection.
     """
-    # TODO: Break show_balances into show_eth_balance and show_token_balance
-
-    if not (provider_uri or wallet):
-        raise ValueError("Provider URI or wallet must be provided to select an account.")
-
-    if not provider_uri:
-        provider_uri = wallet.blockchain.provider_uri
 
-    # Lazy connect the blockchain interface
-    if not BlockchainInterfaceFactory.is_interface_initialized(provider_uri=provider_uri):
-        BlockchainInterfaceFactory.initialize_interface(provider_uri=provider_uri, poa=poa, emitter=emitter)
-    blockchain = BlockchainInterfaceFactory.get_interface(provider_uri=provider_uri)
+    # We use Wallet internally as an account management abstraction
+    if not wallet:
+        if not provider_uri and not signer_uri:
+            raise ValueError("At least a provider URI or signer URI is necessary to select an account")
+        # Lazy connect the blockchain interface
+        if not BlockchainInterfaceFactory.is_interface_initialized(provider_uri=provider_uri):
+            BlockchainInterfaceFactory.initialize_interface(provider_uri=provider_uri, poa=poa, emitter=emitter)
+        wallet = Wallet(provider_uri=provider_uri, signer=Signer.from_signer_uri(signer_uri))
```

Comment:
What happens here if signer_uri is `None`, since we are calling `Signer.from_signer_uri(signer_uri)`? At this point `provider_uri` or `signer_uri` or both are set.

Reply:
Excellent eye 👁️! I was literally fixing this now, since tests were failing for this reason. I'm adding a commit right now with the fix.